nio/client/async_client.py

```diff
@@ -522,39 +563,29 @@ def add_response_callback(
         """
         while True:
             try:
-                responses = []
```

Comment:
It would have been nice if this change was in a separate commit, no need to split it up now but something to keep in mind for the future.